enhancements/installer/openstack-customer-provided-subnets.md

```diff
@@ -0,0 +1,131 @@
+  - "@abhinavdahiya"
+  - "@sdodson"
+creation-date: 2020-03-12
+last-updated: 2020-03-12
+status: planning
+---
+
+# 4.5: OpenStack: Allow Customer Provisioned Subnets
+
+## Release Signoff Checklist
+
+- [ x ] Enhancement is `implementable`
+- [ x ] Design details are appropriately documented from clear requirements
+- [ x ] Test plan is defined
+- [ x ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+In many large organizations, IT groups are separated into teams. A common separation of duties lies between networking/WAN and application or enabling technology deployments. One group will provision or assign OpenStack accounts with core management technology. A second group will be responsible for provisioning networks complete with subnet & zone selection, routing, VPN connections, and ingress/egress rules from the company. Lastly, an applications team (or individual business units) are given a quota for other OpenStack resources to provision application-specific items (instances, containers, databases, pipelines, load balancers, etc.).
+Another example of this split is found in companies that provide public access to their OpenStack resources, giving their tenants/customers quota on compute and storage resources but providing pre-defined networks for them.
+In these scenarios, only OpenStack administrators will manage these networks, which will be generally provider networks, (as opposed to tenant networks), since they require physical network infrastructure configuration (typically VLANs). So when OpenStack tenants log in they will find them pre-created, shared with them and ready to use.
+To accomplish this, our infrastructure needs an OpenStack account access that can be separated into "networking infrastructure" versus "application". Networking infrastructure includes elements related to the networks (networks, subnets, routing tables, internet gateways, NAT, VPN). Application items include OpenStack resources directly touching nodes within the cluster (LBs, security groups, Swift containers, nodes). This is a typical separation of responsibilities among IT groups and deployment/provisioning pipelines within large companies.
+
+## Motivation
+
+### Goals
+As an administrator, I would like to create or reuse my own networks, subnets and routing schemes that I can deploy OpenShift to.
+
+### Non-Goals
+There are no changes in expectations regarding publicly addressable parts of the cluster.
+
+## Proposal
+- Installer allow users to provide a list of subnets that should be used for the cluster. Since there is an expectation that networking is being shared, the installer cannot modify the networking setup, but changes required to the shared resources like Tags that do not affect the behavior for other tenants of the network will be made.
+- The installer validates the assumptions about the networking setup
```

Comment:
i think we have to tread a fine line of what we can easily validate and what our customers do. Our goal is to have zero false negatives here _no customer should fail to deploy just because we think it won't work_

Reply:
That's a good point. I can re-word this to reflect a more cautious position.